includes/Features/Abilities_Explorer/Admin_Page.php

```diff
@@ -0,0 +1,552 @@
+					'invalidJson'   => __( 'Invalid JSON input', 'ai' ),
+					'confirmInvoke' => __( 'Are you sure you want to invoke this ability?', 'ai' ),
+					'copySuccess'   => __( 'Copied to clipboard!', 'ai' ),
+					'copyError'     => __( 'Failed to copy', 'ai' ),
+				),
+			)
+		);
+	}
+
+	/**
+	 * Render the main page.
+	 *
+	 * @since 0.1.0
+	 */
+	public function render_page() {
+		// Check user capabilities
+		if ( ! current_user_can( 'manage_options' ) ) {
+			wp_die( esc_html__( 'You do not have sufficient permissions to access this page.', 'ai' ) );
+		}
+
+		// Get current action
+		$action = isset( $_GET['action'] ) ? sanitize_text_field( wp_unslash( $_GET['action'] ) ) : 'list';
+
+		echo '<div class="wrap ability-explorer-wrap">';
+		echo '<h1>' . esc_html__( 'Ability Explorer', 'ai' ) . '</h1>';
+
+		// Display statistics dashboard
+		if ( 'list' === $action ) {
+			$this->render_statistics();
+		}
+
+		// Render appropriate view based on action
+		switch ( $action ) {
+			case 'view':
+				$this->render_detail_view();
+				break;
+
+			case 'test':
+				$this->render_test_runner();
+				break;
+
+			case 'list':
+			default:
+				$this->render_list_view();
+				break;
+		}
+
+		echo '</div>';
+	}
+
+	/**
+	 * Render statistics dashboard.
+	 *
+	 * @since 0.1.0
+	 */
+	private function render_statistics() {
+		$stats = Ability_Handler::get_statistics();
+
+		?>
+		<div class="ability-explorer-stats">
+			<div class="ability-stat-card">
+				<div class="ability-stat-number"><?php echo esc_html( $stats['total'] ); ?></div>
+				<div class="ability-stat-label"><?php esc_html_e( 'Total Abilities', 'ai' ); ?></div>
+			</div>
+
+			<div class="ability-stat-card">
+				<div class="ability-stat-number"><?php echo esc_html( $stats['by_provider']['Core'] ); ?></div>
+				<div class="ability-stat-label"><?php esc_html_e( 'Core', 'ai' ); ?></div>
+			</div>
+
+			<div class="ability-stat-card">
+				<div class="ability-stat-number"><?php echo esc_html( $stats['by_provider']['Plugin'] ); ?></div>
+				<div class="ability-stat-label"><?php esc_html_e( 'Plugins', 'ai' ); ?></div>
+			</div>
+
+			<div class="ability-stat-card">
+				<div class="ability-stat-number"><?php echo esc_html( $stats['by_provider']['Theme'] ); ?></div>
+				<div class="ability-stat-label"><?php esc_html_e( 'Theme', 'ai' ); ?></div>
+			</div>
+		</div>
+		<?php
+	}
+
+	/**
+	 * Render list view.
+	 *
+	 * @since 0.1.0
+	 */
+	private function render_list_view() {
+		$table = new Ability_Table();
+		$table->prepare_items();
+
+		?>
+		<form method="get">
+			<input type="hidden" name="page" value="ai-abilities-explorer" />
+			<?php
+			$table->search_box( __( 'Search Abilities', 'ai' ), 'ability' );
+			$table->display();
+			?>
+		</form>
+		<?php
+	}
+
+	/**
+	 * Render detail view.
+	 *
+	 * @since 0.1.0
+	 */
+	private function render_detail_view() {
+		$ability_slug = isset( $_GET['ability'] ) ? sanitize_text_field( wp_unslash( $_GET['ability'] ) ) : '';
```

Comment:
This code is duplicated again at line 316. Is it worth calling this once higher up the chain, and then passing the $ability_slug to the relevant render_ method?